main/vstring.c

```diff
@@ -156,13 +156,14 @@ extern void vStringStripNewline (vString *const string)
  */
 extern void vStringStripLeading (vString *const string)
 {
-	while (isspace ((int) string->buffer [0]) && string->length > 0)
+	size_t n = 0;
+
+	while (n < string->length && isspace ((int) string->buffer [n]))
+		n++;
+	if (n > 0)
 	{
-		size_t i;
-		for (i = 1  ;  i < string->length  ;  ++i)
-			string->buffer [i - 1] = string->buffer [i];
-		--string->length;
-		string->buffer [string->length] = '\0';
+		memmove (string->buffer, string->buffer + n, string->length - n);
+		vStringTruncate (string, string->length - n);
```

Comment:
Calling vStringTruncate() is necessary because it sets the length field of the vString structure - it has to be kept up-to-date and correspond to strlen(string->buffer) - other vString functions rely on it. One could also set string->length manually of course instead of calling vStringTruncate() but there would still have to be 2 lines of code. 

This is something where vString differs from g_str-stuff. Having length stored helps performance because when constructing tokens and reading input, parsers add the read characters to the end of the current token one by one and strlen() would have to be called all the time.